### DISCUSSION TITLE
One-to-Many relationship

### DISCUSSION THREAD

**Discussion by zenners** [pocketbase/pocketbase#173] [Q&A] [answered]
Hi I'm trying to get hang of relationships here in PB.

I have a posts and comments collections. I added the post relation to the comments collection. I understand I can do a simple filter on the comments collection by a certain `post.id`

For the sake of learning, I'm trying to retrieve the related comments of a post. From my understanding, applying an API rule on a view would be how to do this "join"

In my posts collection I have this API rule, but can't get the comments. `@collection.comments.post.id = id`

![image](https://user-images.githubusercontent.com/3746914/179938146-1ad8fed2-281b-4c1a-9558-c435a7c420ba.png)

Is this the right approach? or should it be through expand attribute?

**Comment by ganigeorgiev** [comment, marked as answer]
**The API rules don't return additional content to the API responses. They do the opposite - they act as filters and access rules.**

The `@collection.*` rule is used primarily for accessing non-direct relation fields, aka. when you don't have a relation field pointing to the collection.
When you have a relation field, you could just filter directly through it, aka. `post.title = 'something'` - this will return/give access to comment that has a post relation with title `something`. If you use the `expand` parameter on the `comments` collection you'll expand the related `post` record.
If I'm reading correctly, for your use case you want to list all comments associated to a single post. To do that you'll have to query them separately. Here is an example using the JavaScript SDK:

```js
const post = await client.Records.getOne("posts", "MY_POST_ID");

const comments = await client.Records.getFullList("comments", 200, {
  filter: `post.id = "${post.id}"` // when you are checking only the id of the relation, this could be simplified to just 'post = "MY_POST_ID" '
});
```

You could find some additional information for the filters and rules syntax and the difference between `@collection.*`, `@request.*`, etc. here -  https://pocketbase.io/docs/manage-collections/#rules-filters-syntax

**Reply by zenners** [reply]
Thanks for the clarification about collections. Yes that is my use case and my initial assumption. I was just trying to see if it could behave like an ORM. `post.comments`

**Reply by ganigeorgiev** [reply]
You can actually do this also, but you'll have to change where the relation field is placed.

If you create a new relation field `comments` in your `posts` collection and make it to accept multiple values, then you'll be able to fetch a single post with all its comments by doing something like:
```js
const post = await client.Records.getOne("posts", "MY_POST_ID", { expand: "comments" });
```
_* When doing `expand` the view rule of the expanded collection is checked to make sure that the users can see/expand only records that they have access to._
